### PR TITLE
Sync cg and statusdb

### PR DESCRIPTION
**Pull request by islean**
## Description

The database models in cg is out of sync with statusdb. This pr tries to address this using an alembic script and some alterations of the Python models. 

NOTE: This is based on a comparison in stage, not prod

### Added

- Alembic script for changing the database to match the Python models
- Non-nullable constraint on invoice.comment to match db
- Dropped non-nullable constraint on application.min_sequencing_depth

### Changed

- 

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: alembic/versions/526e8250a192_sync_with_db.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "priority",
+        existing_type=mysql.ENUM("research", "standard", "priority", "express", "clinical_trials"),
+        nullable=False,
+    )
+
+    op.create_foreign_key(None, "sample", "invoice", ["invoice_id"], ["id"])
+
+    op.drop_column("sample", "beaconized_at")
+    ####
+    sa.delete(SampleLaneSequencingMetrics).where(SampleLaneSequencingMetrics.flowcell is None)
```
Comment (islean) [inline alembic/versions/526e8250a192_sync_with_db.py:54]: Probably won't work like this, we would need some session + commit but keeping it here as a placeholder for now.

Comment (seallard) [reply alembic/versions/526e8250a192_sync_with_db.py:54]: We could also just drop that table manually

File: cg/store/models.py [hunk 2]
```diff
@@ -728,7 +728,7 @@ class Invoice(Model):
     created_at = Column(types.DateTime, default=dt.datetime.now)
     updated_at = Column(types.DateTime, onupdate=dt.datetime.now)
     invoiced_at = Column(types.DateTime)
-    comment = Column(types.Text)
+    comment = Column(types.Text, nullable=False)
```
Comment (ChrOertlin) [inline cg/store/models.py:731]: I don't think comments are required always. So this column can be NULL

File: alembic/versions/526e8250a192_sync_with_db.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        nullable=False,
+    )
+
+    op.alter_column("invoice", "customer_id", existing_type=mysql.INTEGER(), nullable=False)
+
+    # In the db, the foreign key invoice -> customer does not exist, actually. So below would drop the index in the db
+    # and add a foreign key matching the python model. Foreign keys are indexed by default.
+    op.drop_index("customer_invoice_pk", table_name="invoice")
+    op.create_foreign_key(None, "invoice", "customer", ["customer_id"], ["id"])
+    op.create_foreign_key(None, "pool", "invoice", ["invoice_id"], ["id"])
```
Comment (Vince-janv) [inline alembic/versions/526e8250a192_sync_with_db.py:41]: Very nice! We might need to look trough some code to see that this does not brake anything

File: cg/store/models.py (hunk 2 above)
Comment (Vince-janv) [reply cg/store/models.py:731]: I agree, they can be nullable

**Review by Vince-janv — APPROVED** [review]
Nice to get the db up to date! 🌟

**Comment by islean** [comment]
Will address this in a more up-to-date branch

**Pull request closed without merging**